fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -4841,6 +4917,8 @@ protected boolean checkPossiblyRebuildRecordCounts(@Nonnull RecordMetaData metaD
 
         if (rebuildRecordCounts) {
             // We want to clear all record counts.
+            // This code will leave data behind if the previous RecordCountKey was not grouped
+            // https://github.com/FoundationDB/fdb-record-layer/issues/3335
```

Comment:
Are you planning on tackling that in a separate PR targeted to #3335?

Reply:
Yes, the existence of the state doesn't change things, but I noticed it, and didn't want it to be left unrecorded.